File: frontend/src/app/dtos/assignment/assignment.dto.ts
import { EmployeeRole } from '../../models/employee-role.enum';

export class AssignmentDto {
  productId: number;

  employeeId: number;

  employeeRole: EmployeeRole;

  assigned_date: string;

  end_date?: string | null;

  status: 'Active' | 'Inactive';

  constructor(data: any) {
    this.productId = data.productId;
    this.employeeId = data.employeeId;
    this.employeeRole = data.employeeRole;
    this.assigned_date = data.assigned_date;
    this.end_date = data.end_date;
    this.status = data.status;
  }
}

File: backend/src/main/java/com/example/ecomanager/dtos/AssignmentDTO.java
package com.example.ecomanager.dtos;

import com.example.ecomanager.models.Employee;
import com.example.ecomanager.models.Product;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.time.LocalDate;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class AssignmentDTO {

    private Long productId;

    private Long employeeId;

    private String employeeRole;

    @JsonProperty("assigned_date")
    private LocalDate assignedDate;

    @JsonProperty("end_date")
    private LocalDate endDate;

    @JsonProperty("status")
    private String status;

}
